tron/core/actionrun.py

```diff
@@ -148,8 +148,19 @@ def state_data(self):
 
     @classmethod
     def from_state(cls, state_data):
-        state_data["command_config"] = action.ActionCommandConfig(**state_data["command_config"])
-        return cls(**state_data)
+        # it's possible that we've rolled back to an older Tron version that doesn't support data that we've persisted
+        # (e.g., new fields for an ActionCommandConfig) so ensure that we only load what we currently support
+        valid_command_config_entries_from_state = {
+            field.name: state_data["command_config"][field.name]
+            for field in fields(action.ActionCommandConfig)
```

Comment:
TIL fields()

Reply:
yup, `dataclasses`-only, but it was nice :D (I was also surprised to see that we're already using it in this file too )